sdk/core/azure-core/src/main/java/com/azure/core/implementation/ResponseConstructorsCache.java

```diff
@@ -96,7 +95,8 @@
                         responseStatusCode,
                         responseHeaders));
                 } catch (IllegalAccessException | InvocationTargetException | InstantiationException e) {
-                    throw logger.logExceptionAsError(Exceptions.propagate(e));
+                    throw logger.logExceptionAsError(new RuntimeException("Failed to deserialize 3-parameter"
```

Comment:
You're changing the behaviour here by not using Exceptions.propagate(e). Same with the others below.

```java
throw logger.logExceptionAsError(Exceptions.propagate(
    new RuntimeException("Failed to deserialize 3-parameter response", e)));
```

Reply:
Trying to log the paramCount for debugging purpose.